src/Compilers/CSharp/Portable/CodeGen/EmitExpression.cs

```diff
@@ -2707,27 +2732,13 @@ private void EmitConstantExpression(TypeSymbol type, ConstantValue constantValue
         private void EmitInitObj(TypeSymbol type, bool used, SyntaxNode syntaxNode)
         {
             if (used)
-            {
-                if (type.IsPointerType() || type.SpecialType == SpecialType.System_UIntPtr)
```

Comment:
>if (type.IsPointerType() || type.SpecialType == SpecialType.System_UIntPtr) [](start = 16, length = 75)

It looks like many consumers of EmitInitObj will not be able to take advantage of this optimization. Is this intentional for all of them?